### PR TITLE
rename JULIA_HOME => Sys.BINDIR

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

File: contrib/mac/juliarc.jl [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 # Set up environment for Julia OSX binary distribution
 let
-    ROOT = abspath(JULIA_HOME,"..")
-    ENV["PATH"]="$JULIA_HOME:$(ENV["PATH"])"
+    ROOT = abspath(Sys.BINDIR,"..")
+    ENV["PATH"]="$Sys.BINDIR:$(ENV["PATH"])"
```
Comment (ararslan) [inline contrib/mac/juliarc.jl]: The interpolation here will get `Sys` but not `BINDIR`; you'll need to put parentheses around `Sys.BINDIR`.

**Comment by StefanKarpinski** [comment]
This is pretty messy. I cannot figure out how to get deprecation from Base into Sys.Base to play nicely with the these changes. So this prints lots of "precompile might be broken" warnings during CI. @vtjnash, any ideas how to do this better? I've tried every combination of eval and placement I can think of.

**Comment by StefanKarpinski** [comment]
Ok, screw the deprecations on this what – what a mess. The change itself is easy, making the deprecations not barf in all circumstances seems to be impossible, so I'm just not going to bother.

File: base/deprecated.jl [hunk 2]
```diff
@@ -123,11 +123,10 @@ end
 deprecate(m::Module, s::Symbol, flag=1) = ccall(:jl_deprecate_binding, Void, (Any, Any, Cint), m, s, flag)
 
 macro deprecate_binding(old, new, export_old=true, dep_message=nothing)
+     dep_message == nothing && (dep_message = ", use $new instead")
```
Comment (ararslan) [inline base/deprecated.jl]: Ah, the elusive 5 space indent...

**Comment by ararslan** [comment]
Seems like you should be able to do the renaming and all of that, then just add a `@deprecate_binding`? Not sure though.

**Comment by StefanKarpinski** [comment]
You would think so. But the combination of how this relates to code loading – variable definitions need to happen at a very particular point in the loading process – and the fact that the variable is not just moved inside of Base but moved into another module makes this a nightmare. There may be a way to do it, but I've already wasted enough time on this.

**Comment by ararslan** [comment]
LGTM apart from the one 5-space indent. Nice work and thanks for going through it!

**Comment by StefanKarpinski** [comment]
Why did someone cancel the Circle CI run?

**Review by ararslan — APPROVED** [review]

**Pull request merged**